pkg/plugins/generic/generic_plugin.go

```diff
@@ -186,13 +186,24 @@ func needDrainNode(desired sriovnetworkv1.Interfaces, current sriovnetworkv1.Int
 		for _, iface := range desired {
 			if iface.PciAddress == ifaceStatus.PciAddress {
 				configured = true
-				if iface.NumVfs != ifaceStatus.NumVfs || iface.Mtu != ifaceStatus.Mtu {
+				if iface.EswitchMode == sriovnetworkv1.ESWITCHMODE_SWITCHDEV {
```

Comment:
Add check for ClusterType is openshift

Reply:
I think for vanilla k8s, we also want to use a systemd service to configure the switch device. So from the generic plugin perspective, the logic shall be the same.